Library/Homebrew/cmd/update.rb

```diff
@@ -57,6 +58,7 @@ def update
     else
       puts "Updated Homebrew from #{master_updater.initial_revision[0,8]} to #{master_updater.current_revision[0,8]}."
       report.dump
+      Descriptions.refresh_cache
```

Comment:
Two clarifications:

1. The act of refreshing the cache eats on the order of 5 seconds, give or take, on a reasonably-modern system. It does this on every run, because I couldn’t devise a way of incrementally updating the cache that was both simple and reliable.

2. `Descriptions.refresh_cache` only regenerates the cache if the cache already exists. Users who never invoke `brew desc` or `brew search --desc` shouldn’t see any significant changes in their `brew update` run-time. (Those who did perform a search at some point in the past, but decide that they’d rather not pay for it with every subsequent `brew update`, can use `brew desc --delete-cache` to restore the earlier behavior.)